crates/frontend/src/components/default_config_form/default_config_form.rs

```diff
@@ -48,17 +48,18 @@ where
             },
         );
 
-    let handle_select_dropdown_option = move |selected_function: FunctionsName| {
-        set_function_name.update(|value| {
-            let function_name = selected_function.clone();
-            leptos::logging::log!("function selected: {:?}", function_name);
-            let fun_name = match function_name.as_str() {
-                "None" => None,
-                _ => Some(json!(function_name)),
-            };
-            *value = fun_name;
+    let handle_select_dropdown_option =
+        Callback::new(move |selected_function: FunctionsName| {
+            set_function_name.update(|value| {
+                let function_name = selected_function.clone();
+                leptos::logging::log!("function selected: {:?}", function_name);
+                let fun_name = match function_name.as_str() {
```

Comment:
Can we use if let some here ? 
We only have two options here , don't need to use match here.